Mathlib/Tactic/Linter/DeprecatedSyntaxLinter.lean

```diff
@@ -62,11 +62,40 @@ register_option linter.style.admit : Bool := {
   descr := "enable the admit linter"
 }
 
+/-- The option `linter.style.maxHeartbeats` of the deprecated syntax linter flags usages of
+`set_option maxHeartbeats n in cmd` that do not add a comment explaining the reason for the
+modification of the maxHeartbeats. -/
+register_option linter.style.maxHeartbeats : Bool := {
+  defValue := false
+  descr := "enable the maxHeartbeats linter"
+}
+
+/-- If the input syntax is of the form `set_option <option> num in <string> cmd`,
+where `<option>` contains `maxHeartbeats`, then it returns
+* the `<option>` as a name (typically, `maxHeartbeats` or `synthInstance.maxHeartbeats`);
+* the number `num` and
+* whatever is in `<string>`.
+Note that `<string>` can only consist of whitespace and comments.
+
+Otherwise, it returns `none`.
+-/
+def getSetOptionMaxHeartbeatsComment : Syntax → Option (Name × Nat × Substring)
+  | stx@`(command|set_option $mh $n:num in $_) =>
+    let opt := mh.getId
+    if !opt.components.contains `maxHeartbeats then
+      none
+    else
+      if let some inAtom := stx.find? (·.getAtomVal == "in") then
+        inAtom.getTrailing?.map (opt, n.getNat, ·)
+      else
+        some default
```

Comment:
This branch should never be hit, right?

Reply:
I agree with you: this branch should be unreachable.  I added a comment.